Tests/NIOHTTP2Tests/TestUtilities.swift

```diff
@@ -193,6 +256,53 @@ extension EmbeddedChannel {
     }
 }
 
+extension NIOAsyncTestingChannel {
+    /// This function attempts to obtain a HTTP/2 frame from a connection. It must already have been
+    /// sent, as this function does not call `interactInMemory`. If no frame has been received, this
+    /// will call `XCTFail` and then throw: this will ensure that the test will not proceed past
+    /// this point if no frame was received.
+    func assertReceivedFrame(file: StaticString = #filePath, line: UInt = #line) async throws -> HTTP2Frame {
+        guard let frame: HTTP2Frame = try await assertNoThrowWithValue(await self.readInbound()) else {
+            XCTFail("Did not receive frame", file: (file), line: line)
+            throw NoFrameReceived()
+        }
+
+        return frame
+    }
+
+    /// Asserts that the connection has not received a HTTP/2 frame at this time.
+    func assertNoFramesReceived(file: StaticString = #filePath, line: UInt = #line) async {
+        let content: HTTP2Frame? = try? await assertNoThrowWithValue(await self.readInbound())
+        XCTAssertNil(content, "Received unexpected content: \(content!)", file: (file), line: line)
+    }
+
+    /// Retrieve all sent frames.
+    func sentFrames(file: StaticString = #filePath, line: UInt = #line) async throws -> [HTTP2Frame] {
+        var receivedFrames: [HTTP2Frame] = Array()
+
+        while let frame = try await assertNoThrowWithValue(await self.readOutbound(as: HTTP2Frame.self), file: (file), line: line) {
+            receivedFrames.append(frame)
+        }
+
+        return receivedFrames
+    }
+
+    /// Retrieve all sent frames.
+    func decodedSentFrames(file: StaticString = #filePath, line: UInt = #line) async throws -> [HTTP2Frame] {
+        var receivedFrames: [HTTP2Frame] = Array()
+
+        var frameDecoder = HTTP2FrameDecoder(allocator: self.allocator, expectClientMagic: false)
+        while let buffer = try await assertNoThrowWithValue(await self.readOutbound(as: ByteBuffer.self), file: (file), line: line) {
+            frameDecoder.append(bytes: buffer)
+            if let (frame, _) = try frameDecoder.nextFrame() {
+                receivedFrames.append(frame)
+            }
+        }
+
+        return receivedFrames
+    }
```

Comment:
Are these used anywhere?

Reply:
Not yet, I just ported them over when I did the others in case they were useful but I'll delete them for now and pull them over when needed